crypto/fipsmodule/cipher/cipher.c

```diff
@@ -322,15 +322,15 @@ int EVP_EncryptUpdate(EVP_CIPHER_CTX *ctx, uint8_t *out, int *out_len,
 }
 
 int EVP_EncryptFinal_ex(EVP_CIPHER_CTX *ctx, uint8_t *out, int *out_len) {
-  int n, ret = 0;
+  int n, ret = 0, custom_ret;
   unsigned int i, b, bl;
 
   if (ctx->cipher->flags & EVP_CIPH_FLAG_CUSTOM_CIPHER) {
-    ret = ctx->cipher->cipher(ctx, out, NULL, 0);
-    if (ret < 0) {
-      return 0;
+    custom_ret = ctx->cipher->cipher(ctx, out, NULL, 0);
```

Comment:
You may want to add a comment that this `cipher()` pointer is always a static function which won't contain Service Indicator updates (if I understood our discussion correctly). I see you added that comment in DecryptFinal. I think it's needed in EncryptFinal as well, beside here.